tests/test_internals/test_si.py

```diff
@@ -700,12 +700,14 @@ def test_non_iterative_write(self):
         dtype = self.test_recording_extractor.get_dtype()
         element_size_in_bytes = dtype.itemsize
         num_channels = self.test_recording_extractor.get_num_channels()
+        num_frames = self.test_recording_extractor.get_num_frames()
 
         available_memory_in_bytes = psutil.virtual_memory().available
         frame_size_in_bytes = element_size_in_bytes * num_channels
-        available_memory_in_bytes_plus_one_frame = available_memory_in_bytes + frame_size_in_bytes
 
-        num_frames_to_overflow = available_memory_in_bytes_plus_one_frame / (element_size_in_bytes * num_channels)
+        excess_in_bytes = frame_size_in_bytes * int(num_frames / 2)  # 50 % excess frames
```

Comment:
Doesn't this result in fewer total frames in the mock?

Reply:
excess_in_bytes is what is on top of the estimated available memory, there is a sum down there.

Reply:
Oh, I see - you didn't originally have the `excess_in_bytes` term in the overflow equation.